lib/src/rules/unnecessary_getters_setters.dart

```diff
@@ -89,21 +90,21 @@ class _Visitor extends SimpleAstVisitor<void> {
     }
 
     // Only select getters with setter pairs
-    getters.keys
-        .where((id) => setters.keys.contains(id))
-        .forEach((id) => _visitGetterSetter(getters[id], setters[id]));
+    for (var id in getters.keys.where((id) => setters.keys.contains(id))) {
+      _visitGetterSetter(getters[id], setters[id]);
+    }
   }
 
   void _visitGetterSetter(
       MethodDeclaration? getter, MethodDeclaration? setter) {
-    if (getter != null &&
-        setter != null &&
-        isSimpleSetter(setter) &&
+    if (getter == null || setter == null) return;
```

Comment:
Do we need this test if we've already guaranteed that both maps contain the id?

Better yet, can we remove the `where` on line 93 in favor of this condition (given that `[]` is nullable and we need the test to get around the type system)?

Even better, can we remove both the `where` _and_ this line by writing
```
var getterElement = getter?.declaredElement;
var setterElement = setter?.declaredElement;
```
and testing for `null` on line 103?

Reply:
Oh ho, good trick. Refactored.